VideoLocker/src/main/java/org/edx/mobile/model/course/CourseStructureV1Model.java

```diff
@@ -17,21 +17,20 @@
     @SerializedName("root")
     public String root;
 
-    public  BlockModel getBlockById(String id){
+    public BlockModel getBlockById(String id) {
         return blockData.get(id);
     }
 
     /**
      * if parent is null, returns empty list.
      */
-    public List<BlockModel> getDescendants(BlockModel parent){
+    public List<BlockModel> getDescendants(BlockModel parent) {
         List<BlockModel> descendants = new ArrayList<>();
-        if (parent == null || !parent.isContainer() )
+        if (parent == null || !parent.isContainer() || null == parent.descendants)
```

Comment:
I don't think it's inconsistent state; just because a block type is a "container type" does not mean it *must* have descendants, it just means that it *might* have descendants.

It's just that the server is serializing an empty list as `null` instead of `[]`. I think this is reasonable behaviour for the server (as in, not really a bug) and that the app should treat them the same.